packages/browser/src/notifier.js

```diff
@@ -29,49 +31,69 @@ const pluginStripQueryString = require('@bugsnag/plugin-strip-query-string')
-  if (typeof opts === 'string') opts = { apiKey: opts }
-
-  // configure a client with user supplied options
-  const bugsnag = new Client(opts, schema, { name, version, url })
-
-  // set delivery based on browser capability (IE 8+9 have an XDomainRequest object)
-  bugsnag._setDelivery(window.XDomainRequest ? dXDomainRequest : dXMLHttpRequest)
-
-  // add browser-specific plugins
-  bugsnag.use(pluginDevice)
-  bugsnag.use(pluginContext)
-  bugsnag.use(pluginRequest)
-  bugsnag.use(pluginThrottle)
-  bugsnag.use(pluginSession)
-  bugsnag.use(pluginIp)
-  bugsnag.use(pluginStripQueryString)
-  bugsnag.use(pluginWindowOnerror)
-  bugsnag.use(pluginUnhandledRejection)
-  bugsnag.use(pluginNavigationBreadcrumbs)
-  bugsnag.use(pluginInteractionBreadcrumbs)
-  bugsnag.use(pluginNetworkBreadcrumbs)
-  bugsnag.use(pluginConsoleBreadcrumbs)
-
-  // this one added last to avoid wrapping functionality before bugsnag uses it
-  bugsnag.use(pluginInlineScriptContent)
-
-  bugsnag._logger.debug('Loaded!')
-
-  return bugsnag._config.autoTrackSessions
-    ? bugsnag.startSession()
-    : bugsnag
-}
+const Bugsnag = {
+  _client: null,
+  createClient: (opts) => {
+    // handle very simple use case where user supplies just the api key as a string
+    if (typeof opts === 'string') opts = { apiKey: opts }
+    if (!opts) opts = {}
+
+    // configure a client with user supplied options
+    const bugsnag = new Client(opts, schema, { name, version, url })
+
+    // set delivery based on browser capability (IE 8+9 have an XDomainRequest object)
+    bugsnag._setDelivery(window.XDomainRequest ? dXDomainRequest : dXMLHttpRequest)
+
+    // add browser-specific plugins
+    bugsnag.use(pluginDevice)
+    bugsnag.use(pluginContext)
+    bugsnag.use(pluginRequest)
+    bugsnag.use(pluginThrottle)
+    bugsnag.use(pluginSession)
+    bugsnag.use(pluginIp)
+    bugsnag.use(pluginStripQueryString)
+    bugsnag.use(pluginWindowOnerror)
+    bugsnag.use(pluginUnhandledRejection)
+    bugsnag.use(pluginNavigationBreadcrumbs)
+    bugsnag.use(pluginInteractionBreadcrumbs)
+    bugsnag.use(pluginNetworkBreadcrumbs)
+    bugsnag.use(pluginConsoleBreadcrumbs)
 
-// Angular's DI system needs this interface to match what is exposed
-// in the type definition file (types/bugsnag.d.ts)
-module.exports.Bugsnag = {
-  Client,
-  Event,
-  Session,
-  Breadcrumb
+    // this one added last to avoid wrapping functionality before bugsnag uses it
+    bugsnag.use(pluginInlineScriptContent)
+
+    bugsnag._logger.debug('Loaded!')
+
+    return bugsnag._config.autoTrackSessions
+      ? bugsnag.startSession()
+      : bugsnag
+  },
+  init: (opts) => {
+    if (Bugsnag._client) {
+      Bugsnag._client._logger.warn('Bugsnag.init() was called more than once. Ignoring.')
+      return Bugsnag._client
+    }
+    Bugsnag._client = Bugsnag.createClient(opts)
+    Bugsnag._client._depth += 1
+    return Bugsnag._client
+  }
 }
 
+reduce(keys(Client.prototype), (accum, m) => {
```

Comment:
This might be worthy of a brief comment to aid understanding. Since `Bugsnag` is being mutated and the final accumulated value is never used, it is behaving more like `forEach` than `reduce`.

An alternative to consider:

```
map(keys(Client.prototype), (m) => {
  if (/^_/.test(m)) return
  Bugsnag[m] = function () {
    if (!Bugsnag._client) return console.log(`Bugsnag.${m}() was called before Bugsnag.init()`)
    return Bugsnag._client[m].apply(Bugsnag._client, arguments)
  }
})
```